plugin/src/main/java/org/opennms/plugins/cloud/rest/CloudConfigRestServiceImpl.java

```diff
@@ -43,9 +43,16 @@ public CloudConfigRestServiceImpl(final ConfigurationManager cm) {
 
     @Override
     public Response putActivationKey(final String keyJson) {
-        String key = extractKey(keyJson);
-        this.cm.initConfiguration(key);
-        this.cm.configure();
+        try {
+            String key = extractKey(keyJson);
+            this.cm.initConfiguration(key);
+            this.cm.configure();
+        } catch (Exception e) {
+            return Response
+                    .status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity("{\"status\":\"" + cm.getStatus() + "\", \"message\":\"" + e.getMessage() + "}")
```

Comment:
may be use String.format or JSONObject to make it easier to read.

Reply:
good point. I replaced it with String.format (didn't want to get into the JSon dependency hell just yet :-D